### PR TITLE
FOGL-2322 system test for ingestion of data in a Kafka system

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

File: tests/system/python/test_e2e_kafka.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            assert False, "kafka plugin installation failed"
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        data = {"name": task_name,
+                "plugin": "{}".format(NORTH_PLUGIN_NAME),
+                "type": "north",
+                "schedule_type": 3,
+                "schedule_day": 0,
+                "schedule_time": 0,
+                "schedule_repeat": 30,
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_kafka.py]: is a repeat needed? are we checking with repeats?

File: tests/system/python/README.rst [hunk 2]
```diff
@@ -97,9 +108,41 @@ The minimum required parameters to run these tests are ::
...
+
+   <a href="https://github.com/foglamp/foglamp-north-kafka#build" target="_blank">Build</a>
+
+.. |Confluent| raw:: html
+
+   <a href="https://www.confluent.io/download/" target="_blank">Confluent</a>
+
+.. |Confluent CLI| raw:: html
+
+   <a href="https://docs.confluent.io/current/cli/command-reference/index.html" target="_blank">Confluent CLI</a>
```
[the remark this answers is not in the document]
Comment (ashish-jabble) [reply tests/system/python/README.rst]: No separate installer for CLI, this is reference for to use `confluent commands`

File: tests/system/python/test_e2e_kafka.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+        # Create a consumer for JSON data, starting at the beginning of the topic's
+        # The consumer group is called "CONSUMER_GROUP" and the instance is "CONSUMER_INSTANCE".
+        self._verify_kafka_consumer_group_and_instance(conn)
+
+        # Subscribe the consumer to a topic
+        self._verify_consumer_subscription_to_topic(conn, topic)
+
+        # Then consume some data from a topic using the base URL in the first response.
+        self._verify_consume_data_from_topic(conn)
```
Comment (monika-Dianomic) [inline tests/system/python/test_e2e_kafka.py]: recreate the connection inside _verify_consume_data_from_topic, pass only the host and rest_port!

and make sure to pass header as in DELETE and POST else you will see de-serialization error

Comment (monika-Dianomic) [reply tests/system/python/test_e2e_kafka.py]: browser request: 

http://127.0.0.1:8082/consumers/foglamp_consumer/instances/foglamp_instance/records

response: 
`{"error_code":40601,"message":"The requested embedded data format does not match the deserializer for this consumer instance"}`

File: tests/system/python/test_e2e_kafka.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        conn.request("GET", '/consumers/{}/instances/{}/records'.format(CONSUMER_GROUP, CONSUMER_INSTANCE))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        # TODO assert {'sensor': SENSOR_VALUE} in val
+
+    def _close_consumer(self, conn):
+        # FIXME: This should execute after consumer group/instance creation if in between fails on assertion
+        header = {'Content-Type': 'application/vnd.kafka.v2+json', 'Accept': 'application/vnd.kafka.json.v2+json'}
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_kafka.py]: delete this and use `HEADER` but most importnat is to have it as a teardown (always delete) and may be remove assert

File: tests/system/python/test_e2e_kafka.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+        val = json.loads(r)
+        assert CONSUMER_INSTANCE == val["instance_id"]
+
+    def _verify_consumer_subscription_to_topic(self, conn, topic):
+        data = {"topics": [topic]}
+        conn.request("POST", '/consumers/{}/instances/{}/subscription'.format(CONSUMER_GROUP, CONSUMER_INSTANCE),
+                     json.dumps(data), headers=HEADER)
+        r = conn.getresponse()
+        # No content response
+        assert 204 == r.status
```
Comment (praveen-garg) [inline tests/system/python/test_e2e_kafka.py]: r = r.read().decode() may resolve conn release issue, but its good if you recreate connection in next request and leave here as is, as status is 204.

**Pull request merged**